### PR TITLE
#234 check port before starting

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**

### CODE REVIEW

**Review by dinhlongviolin1 — APPROVED** [review]
Nice one

**Review by dinhlongviolin1 — APPROVED** [review]
Nicely done

**Comment by FredLL-Avaiga** [comment]
The exception is falsely raised when use_reloader is True and the reloader kicks in :-(
Any idea @dinhlongviolin1 ?

**Comment by dinhlongviolin1** [comment]
> The exception is falsely raised when use_reloader is True and the reloader kicks in :-(
> Any idea @dinhlongviolin1 ?

I have no idea. But if it is because of the reloader, you could use the `is_running_from_reloader` guard to check if it is running with a reloader and only perform the port check outside of the reloader.

File: taipy/gui/server.py [hunk 1]
```diff
@@ -230,6 +231,13 @@ def _run_notebook(self):
 
     def runWithWS(self, host, port, debug, use_reloader, flask_log, run_in_thread, ssl_context):
         host_value = host if host != "0.0.0.0" else "localhost"
+        if debug and not is_running_from_reloader():
+            # Check that the port is not already opened
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            result = sock.connect_ex((host_value,port))
```
Comment (FabienLelaquais) [inline taipy/gui/server.py:237]: No complaints from black with the space missing after the comma?

**Review by FabienLelaquais — APPROVED** [review]
This should go into 1.0 as well I'm afraid

File: taipy/gui/server.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply taipy/gui/server.py:237]: nope

**Pull request merged**